File: app/src/main/java/com/android/example/github/AuthenticatorActivity.java
package com.android.example.github;

import android.app.Activity;
import android.os.Bundle;

import com.amazonaws.mobile.auth.ui.SignInUI;
import com.amazonaws.mobile.client.AWSMobileClient;

public class AuthenticatorActivity extends Activity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_authenticator);

        AWSMobileClient.getInstance().initialize(this, awsStartupResult -> {
            SignInUI signin = (SignInUI) AWSMobileClient
                    .getInstance()
                    .getClient(AuthenticatorActivity.this, SignInUI.class);
            signin.login(AuthenticatorActivity.this, MainActivity.class).execute();
        }).execute();
    }
}

File: app/src/main/java/com/MapPost/AuthenticatorActivity.kt
package com.MapPost

import android.app.Activity
import android.os.Bundle

import com.amazonaws.mobile.auth.ui.SignInUI
import com.amazonaws.mobile.client.AWSMobileClient

class AuthenticatorActivity : Activity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_authenticator)

        AWSMobileClient.getInstance().initialize(this) {
            val signin = AWSMobileClient
                    .getInstance()
                    .getClient(this@AuthenticatorActivity, SignInUI::class.java) as SignInUI
            signin.login(this@AuthenticatorActivity, MainActivity::class.java).execute()
        }.execute()
    }
}